### ISSUE TITLE
README

### ISSUE THREAD

**Issue by GriMacez** [GriMacez/-#5]
# 温度预测系统 README

## 一、项目概述
本项目是一个基于长短期记忆网络（LSTM）的温度预测系统，通过历史温度数据以及可能的相关气象数据（如气压、相对湿度），对未来一段时间的温度进行预测。该系统具备数据加载与预处理、模型构建、训练、评估以及未来温度预测等功能，并能可视化训练过程和预测结果。

## 二、功能特性
1. **数据处理**：支持 CSV 和 Excel 格式的数据文件，能自动处理缺失值，按时间排序数据，并提取相关特征。
2. **模型构建**：使用 Keras 构建 LSTM 模型，包含多层 LSTM 单元和 Dropout 层，以防止过拟合。
3. **训练与评估**：使用早停策略避免过拟合，训练过程中记录损失和平均绝对误差（MAE），并在测试集上评估模型性能。
4. **可视化**：绘制训练历史（损失和 MAE）、预测结果以及未来温度预测图。
5. **未来预测**：根据训练好的模型，预测未来指定小时数的温度。

## 三、环境要求
- Python 3.x
- 相关 Python 库：
    - numpy
    - pandas
    - matplotlib
    - scikit-learn
    - tensorflow

你可以使用以下命令安装所需库：
```bash
pip install numpy pandas matplotlib scikit-learn tensorflow
```

### 1. 数据准备
确保你的数据文件包含以下两列：
- `DateTime`：日期时间列，格式需能被 `pandas.to_datetime` 解析。
- `T (degC)`：温度列，单位为摄氏度。

可选列：
- `p (mbar)`：气压列，单位为毫巴。
- `rh (%)`：相对湿度列，单位为百分比。

### 2. 修改数据路径
打开 `114LSTM.py` 文件，将主函数中的 `data_path` 变量替换为你的实际数据文件路径：
```python
if __name__ == "__main__":
    # 请替换为实际数据路径
    data_path = "data.csv"
```

### 3. 运行代码
在终端中运行以下命令启动程序：
```bash
python 114LSTM.py
```

### 4. 配置参数
你可以在 `TemperaturePredictor` 类的初始化函数中调整以下参数：
- `sequence_length`：输入序列的长度（小时），默认为 24。
- `prediction_steps`：预测的步数（小时），默认为 12。

在 `train_model` 方法中，你可以调整以下训练参数：
- `epochs`：训练的轮数，默认为 50。
- `batch_size`：批次大小，默认为 32。
- `patience`：早停策略的耐心值，默认为 10。

## 五、代码结构
- `TemperaturePredictor` 类：主要的温度预测类，包含以下方法：
    - `__init__`：初始化类的属性。
    - `load_data`：加载并预处理数据。
    - `create_sequences`：创建用于 LSTM 模型的序列数据。
    - `prepare_data`：准备 LSTM 模型所需的数据。
    - `build_model`：构建 LSTM 模型。
    - `train_model`：训练 LSTM 模型，并绘制训练历史图。
    - `plot_training_history`：绘制训练过程中的损失和 MAE 曲线。
    - `evaluate_model`：评估模型性能，并绘制预测结果图。
    - `plot_predictions`：绘制测试集上的实际温度和预测温度对比图。
    - `predict_future`：预测未来指定小时数的温度，并绘制预测图。

## 六、输出结果
- `training_history.png`：训练过程中的损失和 MAE 曲线。
- `temperature_predictions.png`：测试集上的实际温度和预测温度对比图。
- `future_temperature_prediction.png`：未来温度预测图。
- 终端输出：训练数据和测试数据的形状、测试集的 RMSE 和 MAE，以及未来温度预测结果。

## 七、注意事项
- 确保数据文件的格式正确，且包含必要的列。
- 训练时间可能较长，具体取决于数据量和模型复杂度。
- 若数据中存在大量缺失值，可能会影响模型性能，建议提前进行更复杂的缺失值处理。
